plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetReaderConfig.java

```diff
@@ -73,6 +74,7 @@ public ParquetReaderConfig setMaxMergeDistance(DataSize distance)
     }
 
     @NotNull
+    @MinDataSize("1MB")
```

Comment:
Before current PR, misconfiguring this property only led to nearby reads not being merged, now it could potentially lead to order of magnitude more requests to filesystem. This has potentially worse consequences than just a small slowdown.
TestHdfsParquetDataSource is using `ParquetReaderOptions`, so I don't think it is affected by constraints in `ParquetReaderConfig`.
TestParquetPageSkipping can be updated to use the new minimum buffer size, TestHdfsParquetDataSource already covers the scenario that the small value was meant to test.
I also noticed now that the related config and session property don't exist in iceberg and delta, we should add those as well and update relevant docs for hive/iceberg/delta.

Reply:
> TestParquetPageSkipping can be updated to use the new minimum buffer size

will 1MB cover the "Small max-buffer-size allows testing mix of small and large ranges in HdfsParquetDataSource#planRead" comment?

> I also noticed now that the related config and session property don't exist in iceberg and delta, we should add those as well and update relevant docs for hive/iceberg/delta.

session property is not there at all (including hive).
I think the config property is there for delta and iceberg.

Reply:
Okay I forgot that we use ParquetReaderConfig directly in delta and iceberg even though they're in trino-hive package.

> will 1MB cover the "Small max-buffer-size allows testing mix of small and large ranges in HdfsParquetDataSource#planRead" comment?

We can drop that comment as this being unit tested elsewhere and there is no mix and small and large anymore, 1MB does help with triggering the chunked reads logic though.